docs/txp.md

```diff
@@ -0,0 +1,334 @@
+`Address'` and compare it with the hash from `Address`.
+
+### Sums check
+
+Recall that at this step we assume that all inputs are known and
+have been resolved into corresponding unspent outputs of previous
+txs. So for each input we know its value. We then compute sum of all
+inputs and all outputs. If sum of outputs is greater than sum of
+inputs, tx is considered invalid. Otherwise this check succeeds and
+the difference between sum of inputs and sum of outputs is considered
+to be transaction fee.
+
+### Tx size check
+
+We compute transaction size as number of bytes in serialized `TxAux`
+(which contains transaction and its witness). Transaction size limit
+is part of `BlockVersionData`. If the size of transaction is greater
+than the limit from the adopted `BlockVersionData`, transaction is
+invalid.
+
+### Fee check
+
+By this time we already know how much fee the transaction includes. We
+also know transaction's size. Fee check depends on the currently
+adopted `TxFeePolicy` (part of `BlockVersionData`). There are two
+options:
+1. Fee policy is `TxFeePolicyTxSizeLinear`. In this case we need to
+   compute minimal required fee for this transaction. This value is
+   computed using fixed-precision arithmetic (precision is 1e-9) using
+   formula `a + b * txSize`. If minimal fee is negative or is greater
+   than maximal number of coins in the system, transaction is rejected
+   (it probably indicates there is a mistake in currently adopted fee
+   policy). Then we compare actual fee of tx with minimal required fee
+   and if it's greater than or equal to the minimal value, transaction
+   is valid with regards to this check.
+2. Fee policy is `TxFeePolicyUnknown`. In this case the check is
+   skipped (i. e. transaction is considered valid with respect to fee
+   policy). It's questionable which behavior we want in this case, but
+   currently it's implemented this way.
+
+Note: if all unspent outputs corresponding to inputs of transaction
+have `Redeem` addresses, such transaction doesn't need to adhere to
+fee policy (i. e. it's free to redeem ADA).
+
+### Bootstrap era check
+
+If the block was created during bootstrap era (which is easy to check,
+because every main block contains `SlotId` in which it was created and
+we know when bootstrap era ends from adopted `BlockVersionData`), all
+output addresses must have `BootstrapEraDistr` distribution.
+
+### GState modifications
+
+If all checks above pass, we modify GState appropriately. We modify
+stakes and UTXO. UTXO modification is trivial:
+* All `TxInUtxo` inputs with are deleted from UTXO.
+* For each output `out` with index `i` we put `(TxInUtxo (hash tx) i,
+  out)` into UTXO.
+
+Stakes modification is a bit more complex, but is also very
+intuitive. We have `TxOut`s corresponding to inputs and outputs of
+transaction. Each `TxOut` can be converted to a list of
+`(StakeholderId, Coin)` pairs. It depends on `AddrStakeDistribution`
```

Comment:
>  It depends on AddrStakeDistribution of given address. For BootstrapEraDistr the value of TxOut is distributed among bootstrap stakeholders proportional to their weights. If distribution is SingleKeyDistr id, then the value of TxOut will be assigned to id. In case of MultiKeyDistr stake will be distributed among multiple stakeholders according to specified portions. 


Bullet list please